go/api/base_client.go

```diff
@@ -0,0 +1,74 @@
+// Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+
+package api
+
+// #cgo LDFLAGS: -L../target/release -lglide_rs
+// #include "../lib.h"
+//
+// void successCallback(uintptr_t channelPtr, char *message);
+// void failureCallback(uintptr_t channelPtr, char *errMessage, RequestErrorType errType);
+import "C"
+
+import (
+	"errors"
+	"unsafe"
+
+	"github.com/aws/glide-for-redis/go/glide/protobuf"
+	"google.golang.org/protobuf/proto"
+)
+
+//export successCallback
+func successCallback(channelPtr C.uintptr_t, cResponse *C.char) {
+	// TODO: Implement when we implement the command logic
+}
+
+//export failureCallback
+func failureCallback(channelPtr C.uintptr_t, cErrorMessage *C.char, cErrorType C.RequestErrorType) {
+	// TODO: Implement when we implement the command logic
+}
+
+type connectionRequestConverter interface {
+	toProtobuf() *protobuf.ConnectionRequest
+}
+
+type baseClient struct {
+	coreClient unsafe.Pointer
+}
+
+func createClient(converter connectionRequestConverter) (*baseClient, error) {
+	request := converter.toProtobuf()
+	msg, err := proto.Marshal(request)
```

Comment:
Do we want to wrap this error with either a different error type or a message?

Reply:
Okay looks to me like the Python client does not wrap the error, it just lets it bubble up. So I think this should be fine